graphqlmetrics/pkg/telemetry/tracing.go

```diff
@@ -0,0 +1,70 @@
+package telemetry
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+)
+
+const (
+	// WithEndpoint respects the following environment variables:
+	// OTEL_EXPORTER_OTLP_ENDPOINT
+	// OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
+	DefaultGrpcTelemetryEndpoint = "cosmo-otel.wundergraph.com"
+)
+
+func buildResourceOptions() resource.Option {
+	return resource.WithAttributes(
+		attribute.String("wg.component.name", DefaultServerName),
+		attribute.String("wg.component.version", serviceVersion),
+	)
+}
+
+func newOtlpTraceGrpcClient() otlptrace.Client {
+	return otlptracegrpc.NewClient(
+		otlptracegrpc.WithEndpoint(DefaultGrpcTelemetryEndpoint),
+		otlptracegrpc.WithCompressor("gzip"),
+	)
+}
+
+func newTraceExporter() (*otlptrace.Exporter, error) {
+	return otlptrace.New(
+		context.Background(),
+		newOtlpTraceGrpcClient(),
+	)
+}
+
+func (c *Config) NewTracerProvider() (*sdktrace.TracerProvider, error) {
+	exporter, err := newTraceExporter()
+	if err != nil {
+		return nil, err
+	}
+
+	resources, err := resource.New(
+		context.Background(),
```

Comment:
`context.Background()` is rarely used. In most of the cases, you should allow to pass a custom context to allow cancellations.

Reply:
I removed tracing related references for now to split up metrics and tracing properly.